interfaces/thirdparty/cp2k.py

```diff
@@ -252,6 +303,107 @@ def get_multigrid_info(self):
 
         return dic
 
+    def get_md_infos(self, file=None, cache=False):
+        """Read the MD-info sections.
+
+        Returns a list with descriptors and a nested list containing the values for each timestep.
+
+        Set ``cache`` to save the results in ``self.md_infos`` to speed up analysis by avoiding I/O.
+        """
+        if hasattr(self, 'md_infos'):
+            return self.md_infos
+        if not file:
+            file = self.job._filename('out')
+        delimiter = '*'*10
+        chunk = self.get_file_chunk(file, begin=delimiter, end=delimiter, match=0, inc_begin=True)
+        frames = [[]]
+        for line in chunk:
+            if not line:
+                continue
+            if delimiter in line:
+                if len(frames[-1]) == 0:
+                    continue
+                frames.append([])
+                continue
+            if not '=' in line:
+                continue
+            l = line.strip().split('=')
+            l = [ x.strip() for x in l ]
+            frames[-1].append(l)
+
+        ret = []
+        for frame in frames:
+            if any('INITIAL' in x[0].upper() for x in frame):
+                continue
+            elif len(frame) == 0:
+                continue
+            else:
+                ret.append(frame)
+
+        names = [ x[0] for x in ret[0] ]
+        for i, frame in enumerate(ret):
+            assert names == [ x[0] for x in frame ], ("Namings in output not constant?")
+            assert len(frame) == len(names), ("{:}".format(frame))
+            assert set([len(x) for x in frame]) == set([2])
```

Comment:
```suggestion
            assert set(len(x) for x in frame) == {2}
```
Nit, feel free to ignore: 
`set()` can take any iterable object as long as its element are hashable. 
This includes both lists and generators.

Reply:
Ahh, didn't know. Thx!